realm/realm-annotations-processor/src/test/resources/io/realm/NullTypesRealmProxy.java

```diff
@@ -203,6 +204,12 @@ private void injectObjectContext() {
             if (!proxyState.getAcceptDefaultValue$realm()) {
                 return;
             }
+            final Row row = proxyState.getRow$realm();
+            if (value == null) {
+                throw new IllegalArgumentException("Trying to set non-nullable field 'fieldStringNotNull' to null.");
+            }
+            row.getTable().setString(columnInfo.fieldStringNotNullIndex, row.getIndex(), value, true);
+            return;
```

Comment:
We need to ignore setting primary key when under constructon in order to allow specifying primary key value for unmanaged objects like:
```
public class Foo extends RealmObject {
    @PrimaryKey
    public String id = UUID.randomUUID().toString();
}

Foo unmanaged = new Foo();
```